core/src/rpc.rs

```diff
@@ -275,144 +327,246 @@ pub struct RpcVersionInfo {
     pub solana_core: String,
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub enum BankType {
+    Recent,
+    Rooted,
+}
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct RpcConfidenceConfig {
+    bank_type: Option<BankType>,
+    confirmations: Option<usize>,
+    percentage: Option<f64>,
+}
+
+fn parse_params(params: Option<Params>) -> ConfidenceConfig {
+    params
+        .and_then(|params| params.parse().ok())
+        .map(|confidence_config: RpcConfidenceConfig| {
+            let mut final_params = ConfidenceConfig::default();
+            if let Some(confirmations) = confidence_config.confirmations {
+                final_params.confirmations = confirmations.min(MAX_LOCKOUT_HISTORY).max(1);
```

Comment:
nit: Can we fail the RPC request with an invalid input error if `confirmations` or `percentage` are out of bounds?